### DISCUSSION TITLE
Cannot read properties of null (reading 'writeFile')

### DISCUSSION THREAD

**Discussion by DidoNael** [requarks/wiki#6725] [Error / Bug Report]
Dear friends, have this problem (Cannot read properties of null (reading 'writeFile'))  using docker even in the most recent or old versions as mentioned (2.5.284) and I believe that it cannot be because of the named folders (server) and (node) because even creating another instance of docker occurs the same problem.
I'm currently on version 2.5.299, here are some images below.

My folder structure starts /root/ where I have other folders just below.

The strangest thing is that anyway it ends up saving the page but it keeps getting these  errors asking to select a local or remote version.

Thanks!

![image](https://github.com/requarks/wiki/assets/52351947/b0867ea9-b7e1-4a2d-a371-f2379f1a0ca0)
![image](https://github.com/requarks/wiki/assets/52351947/d5554c4d-a892-49e3-b8d5-d96205b578c3)
![image](https://github.com/requarks/wiki/assets/52351947/93a0a3f6-2727-4af7-a2f7-bfa0213b2628)
![image](https://github.com/requarks/wiki/assets/52351947/f0c6a58b-a22e-49ee-831d-c1de36d54212)

**Comment by wladsrostova** [comment]
Same problem. Version 2.5.300

**Comment by madodig** [comment]
Hi, you have one of the storage targets enabled (in Admin / Storage). Per logs, it's SFTP. The page is saved (to the wiki DB), but it is not saved to the sync target (SFTP in your case).

Check your SFTP settings, because Wiki needs to be able to authenticate and, depending on the configured sync direction, have read and/or write permissions on the directory configured in Base Directory Path field (that folder needs to exist on the sync target).

> My folder structure starts /root/ where I have other folders just below.

If you are referring to Base Directory Path in SFTP backup target settings, you would not want to use /root here, nor would you want to configure sync authentication with root user. Use a dedicated user for backup and a folder with write permissions for that user (ie: /wiki, /wiki-backup, /home/<configured_user>/wiki, ...).
If you are referring to Wiki hierarchy, while Base Directory Path in SFTP settings is set to /, I guess the same applies.

**Comment by DidoNael** [comment]
Dear friend, your answer was correct, after deactivating SFTP it normalized these messages, in my case I don't need it as I directly back up the database.
Thank you, health.

![image](https://github.com/requarks/wiki/assets/52351947/e2caa4aa-7533-4aee-a976-71efa05295e1)

**Reply by madodig** [reply]
Hi, glad it helped. Can you mark the comment as solution? It might help others in search for an answer. Thanks.

**Comment by hadi2905** [comment]
Hi madodig,
Helped for me too. Like DidoNael I disabled SFTP an the error didn't occur anymore.
Thanks!
